app/controllers/datasets_controller.rb

```diff
@@ -1,4 +1,9 @@
 # frozen_string_literal: true
+
+# rubocop:disable Metrics/ClassLength
+# rubocop:disable Metrics/AbcSize
+# rubocop:disable Metrics/MethodLength
+# rubocop:disable Style/For
```

Comment:
I disabled a few warnings for now since I wanted to dump all the code in a single place for now. If this PR is accepted and the code is merged we'll re-activate the warnings once we start moving some of this code out of the controller.

Reply:
👍🏻